### PR TITLE
Adding URL summary option to DocSum Gradio-UI

### PR DESCRIPTION
**Pull request by MSCetin37**
## Description

Adding URL summary option to DocSum Gradio-UI
 
## Issues

N/A

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)
- [ ] Others (enhancement, documentation, validation, etc.)

### CODE REVIEW

**Comment by MSCetin37** [comment]
@ashahba

**Review by ashahba — APPROVED** [review]
LGTM!

File: DocSum/ui/gradio/docsum_ui_gradio.py [hunk 1]
```diff
@@ -152,6 +189,7 @@ def generate_summary(self, doc_content, document_type="text"):
 
         except requests.exceptions.RequestException as e:
             logger.error("Request exception: %s", e)
+            print("")
```
Comment (lvliang-intel) [inline DocSum/ui/gradio/docsum_ui_gradio.py:192]: why print empty string?

Comment (WenjiaoYue) [reply DocSum/ui/gradio/docsum_ui_gradio.py:192]: Please make sure CI passes and delete useless code.

File: DocSum/ui/gradio/docsum_ui_gradio.py [hunk 2]
```diff
@@ -90,6 +91,38 @@ def read_video_file(self, file):
...
+
+        self.page_content = ""
+
+        logger.info(">>> Reading url: %s", url)
+        if self.is_valid_url(url=url):
+            loader = UnstructuredURLLoader([url])
+            page = loader.load()
+            self.page_content = [content.page_content for content in page][0]
+        else:
+            msg = f"Invalid URL '{url}'. Make sure the link provided is a valid URL.url"
```
Comment (dcmiddle) [inline DocSum/ui/gradio/docsum_ui_gradio.py:125]: typo? URL.url

File: DocSum/ui/gradio/docsum_ui_gradio.py [hunk 3]
```diff
@@ -90,6 +91,38 @@ def read_video_file(self, file):
...
+            str: The content of the website or an error message if the url is unsupported.
+        """
+
+        self.page_content = ""
+
+        logger.info(">>> Reading url: %s", url)
+        if self.is_valid_url(url=url):
+            loader = UnstructuredURLLoader([url])
+            page = loader.load()
+            self.page_content = [content.page_content for content in page][0]
```
Comment (dcmiddle) [inline DocSum/ui/gradio/docsum_ui_gradio.py]: Why take only the first page_content?

File: DocSum/ui/gradio/docsum_ui_gradio.py [hunk 4]
```diff
@@ -91,6 +92,42 @@ def read_video_file(self, file):
...
+
+        self.page_content = ""
+
+        logger.info(">>> Reading url: %s", url)
+        if self.is_valid_url(url=url):
+            os.environ["no_proxy"] = f"{os.environ.get('no_proxy', '')},{url}".strip(",")
+            try:
+                loader = UnstructuredURLLoader([url])
+                page = loader.load()
+                self.page_content = [content.page_content for content in page][0]
```
Comment (dcmiddle) [inline DocSum/ui/gradio/docsum_ui_gradio.py:120]: Why take only the first page_content?

**Comment by dcmiddle** [comment]
Please clean up the commit history. All logically related changes in 1 commit. That means squash several of these commits. Add the reason for the change in the commit comments.
See https://cbea.ms/git-commit/ for other tips on commit messages.

**Comment by dcmiddle** [comment]
Please update the PR description with the `unstructured` dependency

File: DocSum/ui/gradio/docsum_ui_gradio.py (hunk 4 above)
Comment (okhleif-IL) [reply DocSum/ui/gradio/docsum_ui_gradio.py:120]: there is only one page, just adding [0] to extract the text from the list

File: DocSum/ui/gradio/docsum_ui_gradio.py (hunk 2 above)
Comment (MSCetin37) [reply DocSum/ui/gradio/docsum_ui_gradio.py]: fixed

File: DocSum/ui/gradio/docsum_ui_gradio.py (hunk 1 above)
Comment (MSCetin37) [reply DocSum/ui/gradio/docsum_ui_gradio.py]: fixed

**Review by lvliang-intel — APPROVED** [review]

**Review by WenjiaoYue — APPROVED** [review]

**Pull request merged**